From 9d8173e9818673cf11a1d91963fe2e6b284cac7c Mon Sep 17 00:00:00 2001
From: Larry Wu <larry.wu@intel.com>
Date: Wed, 23 Dec 2015 17:20:01 -0700
Subject: [PATCH 249/441] Title: Add device name and class name to Puma ACPI
 platform driver IssueID:   CPM-16 IssueType: ECR Symptom:   Puma ACPI
 platform driver doesn't specify the class and device name. RootCause: N/A.
 Resolution: N/A. Test:      Ran unit tests on functions. Reviewer:  Vinay
 Patel ReviewID:  CCID-119407

---
 drivers/platform/x86/puma_acpi.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

--- a/drivers/platform/x86/puma_acpi.c
+++ b/drivers/platform/x86/puma_acpi.c
@@ -885,9 +885,10 @@ static int puma_acpi_add(struct acpi_dev
    struct puma_acpi_data *data;
    struct task_struct *task;
    int resource_index;
-	int ret;
+   int ret;
    unsigned int power_status;
-
+   char *str;
+   
    data = kzalloc(sizeof(*data), GFP_KERNEL);
    if (!data)
       return -ENOMEM;
@@ -900,6 +901,12 @@ static int puma_acpi_add(struct acpi_dev
       resource_index = PUMA_ACPI_DATA;
       goto puma_acpi_add_error;
    }
+   
+   str = acpi_device_name(acpi_dev);
+   strcpy(str, "PUMA_ACPI_DEVICE");
+   
+   str = acpi_device_class(acpi_dev);
+   strcpy(str, "Platform");
 
    ret = puma_acpi_get_data(data, "GCSR", &power_status);
    if(ret) {
